pkg/channel/distributed/controller/kafkachannel/dispatcher.go

```diff
@@ -144,20 +148,35 @@ func (r *Reconciler) reconcileDispatcherService(ctx context.Context, logger *zap
 		}
 	} else {
 
+		// Determine whether the existing service is different in a way that demands a patch
+		// such as missing required labels or spec differences
+		patch, needsUpdate := util.CheckServiceChanged(logger, existingService, newService)
+
 		// Log Deletion Timestamp & Finalizer State
-		if service.DeletionTimestamp.IsZero() {
+		if existingService.DeletionTimestamp.IsZero() {
 			logger.Info("Successfully Verified Dispatcher Service")
 		} else {
-			if util.HasFinalizer(r.finalizerName(), &service.ObjectMeta) {
+			if util.HasFinalizer(r.finalizerName(), &newService.ObjectMeta) {
 				logger.Info("Blocking Pending Deletion Of Dispatcher Service (Finalizer Detected)")
 			} else {
 				logger.Warn("Unable To Block Pending Deletion Of Dispatcher Service (Finalizer Missing)")
 			}
 		}
 
-		// Return Success
-		return nil
+		// Patch the service in Kubernetes if necessary
+		if needsUpdate {
+			logger.Info("Dispatcher Service Changed - Patching")
+			_, err = r.kubeClientset.CoreV1().Services(existingService.Namespace).Patch(ctx, existingService.Name, types.JSONPatchType, patch, metav1.PatchOptions{})
+			if err == nil {
+				controller.GetEventRecorder(ctx).Event(channel, corev1.EventTypeNormal, event.DispatcherServicePatched.String(), "Dispatcher Service Patched")
+			} else {
+				controller.GetEventRecorder(ctx).Event(channel, corev1.EventTypeWarning, event.DispatcherServicePatchFailed.String(), "Dispatcher Service Patch Failed")
+				channel.Status.MarkDispatcherFailed(event.DispatcherServicePatchFailed.String(), "Failed To Patch Dispatcher Service: %v", err)
+				return err
```

Comment:
everywhere else we create events we also log error/success - can we do the same here?

Reply:
Yes; that was a miss on my part